File: ai-proxy-frontend/src/types/stats.ts
export interface RequestLog {
  id: number;
  request_id: string;
  model_id: number | null;
  model_name: string | null;
  route: "local" | "cloud" | "manual" | null;
  confidence: number | null;
  route_reason: string | null;
  request_excerpt: string | null;
  original_tokens: number | null;
  compressed_tokens: number | null;
  prompt_tokens: number | null;
  completion_tokens: number | null;
  latency_ms: number | null;
  status: "success" | "error" | "cancelled";
  error: string | null;
  created_at: string;
}

export interface LogPage {
  total: number;
  page: number;
  page_size: number;
  items: RequestLog[];
}

export interface StatsOverview {
  today_requests: number;
  today_errors: number;
  today_success_rate: number | null;
  today_avg_latency_ms: number | null;
  total_requests: number;
  tokens_saved_today: number;
  tokens_saved_total: number;
  route_distribution: Record<string, number>;
}

export interface DailySavings {
  date: string;
  requests: number;
  original_tokens: number;
  compressed_tokens: number;
  saved_tokens: number;
}

export interface SavingsTrend {
  days: DailySavings[];
}

export interface LogQuery {
  page: number;
  page_size: number;
  route?: string;
  status?: string;
  model_name?: string;
  days?: number;
}

File: api_gateway/app/schemas/stats.py
"""统计与监控相关的 Pydantic Schema(文档 6.3 节)。"""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class RequestLogOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    request_id: str
    model_id: int | None
    model_name: str | None
    route: str | None
    confidence: float | None
    route_reason: str | None
    request_excerpt: str | None
    original_tokens: int | None
    compressed_tokens: int | None
    prompt_tokens: int | None
    completion_tokens: int | None
    latency_ms: int | None
    status: str
    error: str | None
    created_at: datetime


class LogPage(BaseModel):
    total: int
    page: int
    page_size: int
    items: list[RequestLogOut]


class StatsOverview(BaseModel):
    today_requests: int
    today_errors: int
    today_success_rate: float | None  # 0~1,今日无请求时为 None
    today_avg_latency_ms: int | None
    total_requests: int
    tokens_saved_today: int
    tokens_saved_total: int
    route_distribution: dict[str, int]  # 今日 local/cloud/manual 分布


class DailySavings(BaseModel):
    date: str  # YYYY-MM-DD(服务器本地时区)
    requests: int
    original_tokens: int
    compressed_tokens: int
    saved_tokens: int


class SavingsTrend(BaseModel):
    days: list[DailySavings]
